test/e2e/basic_test.go

```diff
@@ -302,16 +319,19 @@ func TestReconcileGatewayRoutesOnStartup(t *testing.T) {
 		t.Logf("Found all expected gateway routes")
 	}
 
-	routeToDelete := routes[0]
+	var routeToDelete *Route
+	if encapMode.SupportsEncap() {
```

Comment:
Shouldn't we just skip this test altogether for `NoEncap`?
It may also be a good idea to update the comment for this test to explain the differences in the test for the different encap modes.

Reply:
for noencap mode, test excises deleting stale CIDR routes.